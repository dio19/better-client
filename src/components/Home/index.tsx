import React from 'react';
import { useEffect } from "react"
import {
  NavLink
} from 'react-router-dom';
import "./index.scss"

const Home = () => {

  useEffect(() => {
    const script = document.createElement("script")
    script.innerHTML = `
    var swiper = new Swiper(".method-slider", {
      spaceBetween: 30,
      effect: "fade",
      loop: true,
      mousewheel: {
        invert: false,
      },
      // autoHeight: true,
      pagination: {
        el: ".method-slider__pagination",
        clickable: true,
      },
    })`
    document.body.appendChild(script)
  }, [])

  return (
    <div className="container">
            <div className="method-slider">
        <div className="method-slider__wrp swiper-wrapper">
          <div className="method-slider__item swiper-slide">
            <div className="method-slider__img">
              <img
                src="https://blog.hubspot.com/hubfs/customer-iinteraction.jpg"
                alt=""
              />
            </div>
            <div className="method-slider__content">
              <span className="method-slider__method">POST</span>
              <div className="method-slider__title">Add Customer</div>
              <div className="method-slider__text">
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                Recusandae voluptate repellendus magni illo ea animi?{" "}
              </div>
              <NavLink
                  to="/add-customer"
                  children={<div className="method-slider__button">GO TO THE FORM</div>}
              ></NavLink>
            </div>
          </div>
          <div className="method-slider__item swiper-slide">
            <div className="method-slider__img">
              <img
                src="https://blog.hubspot.com/hubfs/Align-Sales-Customer-Service.jpg"
                alt=""
              />
            </div>
            <div className="method-slider__content">
              <span className="method-slider__method">PUT/POST</span>
              <div className="method-slider__title">Update or delete Customer</div>
              <div className="method-slider__text">
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                Recusandae voluptate repellendus magni illo ea animi?
              </div>
              <NavLink
                  to="/update-or-delete"
                  children={<div className="method-slider__button">GO TO THE FORM</div>}
              ></NavLink>
            </div>
          </div>

          <div className="method-slider__item swiper-slide">
            <div className="method-slider__img">
              <img
                src="https://www.hubspot.com/hubfs/customer%20acquisition%20specialist_featured%20image.jpg"
                alt=""
              />
            </div>
            <div className="method-slider__content">
              <span className="method-slider__method">TABLE</span>
              <div className="method-slider__title">All Customers</div>
              <div className="method-slider__text">
                Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                Recusandae voluptate repellendus magni illo ea animi?
              </div>
              <NavLink
                  to="/customers"
                  children={<div className="method-slider__button">GO TO THE TABLE</div>}
              ></NavLink>
            </div>
          </div>
        </div>
        <div className="method-slider__pagination"></div>
      </div>
    </div>

  )
}

export default Home;
