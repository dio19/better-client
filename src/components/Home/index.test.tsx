import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Home from "./index";

describe("BuyLevel6 - Benefits Group Section Mobile", () => {
  //   test('It should render Benefits items component correctly', () => {
  //     const props = {
  //       id: 'benefits-items',
  //       type: 'benefits-items',
  //       content: {
  //         title: 'Los mejores descuentos en envíos',
  //         items: [
  //           {
  //             icon: 'truck_shipping_large',
  //             description: '45% OFF en tus envíos',
  //           },
  //           {
  //             icon: 'truck_large_v2',
  //             description: 'Envíos gratis a partir de $ 4.000',
  //           },
  //         ],
  //       },
  //     };

  //     const { asFragment } = render(<BenefitsComponent {...props} />);
  //     expect(asFragment()).toMatchSnapshot();
  //   });

  test("It should render Benefits image component correctly", () => {
    const { asFragment } = render(
      <Router>
        <Home />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  //   test('It should render Benefits Partners component correctly', () => {
  //     const props = {
  //       id: 'partner-benefits',
  //       type: 'partner-benefits',
  //       content: {
  //         title: 'Más descuentos en suscripciones',
  //         pill: {
  //           label: '50% OFF',
  //           color: '#FFFFFF',
  //           background_color: '#00A650',
  //         },
  //         items: [
  //           {
  //             name: 'HBO Max',
  //             description: 'Grandes estrenos, series exclusivas, contenido para la familia y más.',
  //             picture: {
  //               alt: 'HBO Max',
  //               src: {
  //                 '1x': 'https://http2.mlstatic.com/resources/frontend/statics/loyal/partners/hbomax/logo/logoSquare@2x.png?v=1',
  //                 '2x': 'https://http2.mlstatic.com/resources/frontend/statics/loyal/partners/hbomax/logo/logoSquare@3x.png?v=1',
  //               },
  //             },
  //           },
  //         ],
  //       },
  //     };

  //     const { asFragment } = render(<BenefitsComponent {...props} />);
  //     expect(asFragment()).toMatchSnapshot();
  //   });
});
