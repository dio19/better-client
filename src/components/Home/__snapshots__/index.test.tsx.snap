// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyLevel6 - Benefits Group Section Mobile It should render Benefits image component correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="method-slider"
    >
      <div
        class="method-slider__wrp swiper-wrapper"
      >
        <div
          class="method-slider__item swiper-slide"
        >
          <div
            class="method-slider__img"
          >
            <img
              alt=""
              src="https://blog.hubspot.com/hubfs/customer-iinteraction.jpg"
            />
          </div>
          <div
            class="method-slider__content"
          >
            <span
              class="method-slider__method"
            >
              POST
            </span>
            <div
              class="method-slider__title"
            >
              Add Customer
            </div>
            <div
              class="method-slider__text"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Recusandae voluptate repellendus magni illo ea animi? 
            </div>
            <a
              class=""
              href="/add-customer"
            >
              <div
                class="method-slider__button"
              >
                GO TO THE FORM
              </div>
            </a>
          </div>
        </div>
        <div
          class="method-slider__item swiper-slide"
        >
          <div
            class="method-slider__img"
          >
            <img
              alt=""
              src="https://blog.hubspot.com/hubfs/Align-Sales-Customer-Service.jpg"
            />
          </div>
          <div
            class="method-slider__content"
          >
            <span
              class="method-slider__method"
            >
              PUT/POST
            </span>
            <div
              class="method-slider__title"
            >
              Update or delete Customer
            </div>
            <div
              class="method-slider__text"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Recusandae voluptate repellendus magni illo ea animi?
            </div>
            <a
              class=""
              href="/update-or-delete"
            >
              <div
                class="method-slider__button"
              >
                GO TO THE FORM
              </div>
            </a>
          </div>
        </div>
        <div
          class="method-slider__item swiper-slide"
        >
          <div
            class="method-slider__img"
          >
            <img
              alt=""
              src="https://www.hubspot.com/hubfs/customer%20acquisition%20specialist_featured%20image.jpg"
            />
          </div>
          <div
            class="method-slider__content"
          >
            <span
              class="method-slider__method"
            >
              TABLE
            </span>
            <div
              class="method-slider__title"
            >
              All Customers
            </div>
            <div
              class="method-slider__text"
            >
              Lorem ipsum dolor sit amet consectetur, adipisicing elit. Recusandae voluptate repellendus magni illo ea animi?
            </div>
            <a
              class=""
              href="/customers"
            >
              <div
                class="method-slider__button"
              >
                GO TO THE TABLE
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="method-slider__pagination"
      />
    </div>
  </div>
</DocumentFragment>
`;
